packages/compiler/src/core/checker.ts

```diff
@@ -2978,6 +2996,8 @@ export function createChecker(program: Program): Checker {
       linkType(links, type, mapper);
     }
 
+    checkDirectives(prop, type);
```

Comment:
Could we have that be done automatically in `createType`?

Reply:
I'll try it and see if there's any unintended consequences to it.  It'd certainly be nice if we didn't have to keep track of where all those calls should go.